domain/testing/atomic.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package testing
+
+import (
+	"context"
+
+	"go.uber.org/mock/gomock"
+
+	"github.com/juju/juju/domain"
+)
+
+type testingAtomicContext struct {
+	ctx context.Context
+}
+
+func NewAtomicContext(ctx context.Context) *testingAtomicContext {
+	return &testingAtomicContext{
+		ctx: ctx,
+	}
+}
+
+func (t *testingAtomicContext) Context() context.Context {
+	return t.ctx
+}
+
+var IsAtomicContextChecker = gomock.AssignableToTypeOf(domain.AtomicContext(&testingAtomicContext{}))
```

Comment:
Comment me.

Do we want this here, or instead copy it for every test?

Reply:
Since we now have a `domain/testing/atomic.go` file we may as well include it here, no?